posthog/hogql/expr_parser.py

```diff
@@ -0,0 +1,356 @@
+EVENT_FIELDS = ["id", "uuid", "event", "timestamp", "distinct_id"]
+# "person.*" fields you can select from in the events query
+EVENT_PERSON_FIELDS = ["id", "created_at", "properties"]
+
+# HogQL -> ClickHouse allowed transformations
+CLICKHOUSE_FUNCTIONS = {
+    # arithmetic
+    "abs": "abs",
+    "max2": "max2",
+    "min2": "min2",
+    # type conversions
+    "toInt": "toInt64OrNull",
+    "toFloat": "toFloat64OrNull",
+    "toDecimal": "toDecimal64OrNull",
+    "toDate": "toDateOrNull",
+    "toDateTime": "parseDateTimeBestEffort",
+    "toIntervalSecond": "toIntervalSecond",
+    "toIntervalMinute": "toIntervalMinute",
+    "toIntervalHour": "toIntervalHour",
+    "toIntervalDay": "toIntervalDay",
+    "toIntervalWeek": "toIntervalWeek",
+    "toIntervalMonth": "toIntervalMonth",
+    "toIntervalQuarter": "toIntervalQuarter",
+    "toIntervalYear": "toIntervalYear",
+    "toString": "toString",
+    # date functions
+    "now": "now",
+    "toMonday": "toMonday",
+    "toStartOfYear": "toStartOfYear",
+    "toStartOfQuarter": "toStartOfQuarter",
+    "toStartOfMonth": "toStartOfMonth",
+    "toStartOfWeek": "toStartOfWeek",
+    "toStartOfDay": "toStartOfDay",
+    "toStartOfHour": "toStartOfHour",
+    "toStartOfMinute": "toStartOfMinute",
+    "toStartOfSecond": "toStartOfSecond",
+    "toStartOfFiveMinutes": "toStartOfFiveMinutes",
+    "toStartOfTenMinutes": "toStartOfTenMinutes",
+    "toStartOfFifteenMinutes": "toStartOfFifteenMinutes",
+    "toTimezone": "toTimezone",
+    "age": "age",
+    "dateDiff": "dateDiff",
+    "dateTrunc": "dateTrunc",
+    "formatDateTime": "formatDateTime",
+    # string functions
+    "length": "lengthUTF8",
+    "empty": "empty",
+    "notEmpty": "notEmpty",
+    "leftPad": "leftPad",
+    "rightPad": "rightPad",
+    "lower": "lower",
+    "upper": "upper",
+    "repeat": "repeat",
+    "format": "format",
+    "concat": "concat",
+    "coalesce": "coalesce",
+    "substring": "substringUTF8",
+    "appendTrailingCharIfAbsent": "appendTrailingCharIfAbsent",
+    "endsWith": "endsWith",
+    "startsWith": "startsWith",
+    "trim": "trimBoth",
+    "trimLeft": "trimLeft",
+    "trimRight": "trimRight",
+    "extractTextFromHTML": "extractTextFromHTML",
+    "like": "like",
+    "ilike": "ilike",
+    "notLike": "notLike",
+    "replace": "replace",
+    "replaceOne": "replaceOne",
+    # array functions
+    "tuple": "tuple",
+    # conditional
+    "if": "if",
+    "multiIf": "multiIf",
+    # rounding
+    "round": "round",
+    "floor": "floor",
+    "ceil": "ceil",
+    "trunc": "trunc",
+}
+# Permitted HogQL aggregations
+HOGQL_AGGREGATIONS = [
+    "total",
+    "min",
+    "max",
+    "sum",
+    "avg",
+    "any",
+]
+# Keywords passed to ClickHouse without transformation
+KEYWORDS = ["true", "false", "null"]
+
+# Allow-listed fields returned when you select "*" from events. Person and group fields will be nested later.
+SELECT_STAR_FROM_EVENTS_FIELDS = [
```

Comment:
Let's be conservative here and remove groups-related columns for now.

groups-on-events is not even actively worked on right now.

I'd love it if `SELECT *` wasn't a thing from a perf standpoint but I digess. :)